mantidimaging/gui/windows/image_load_dialog/presenter.py

```diff
@@ -73,66 +75,121 @@ def do_update_sample(self, selected_file: str) -> None:
+
+        :return None
+        """
+        filename_group = FilenameGroup.from_file(Path(selected_file))
+        filename_group.find_shutter_count_file()
+        field.set_images(list(filename_group.all_files()))
+        self._update_field_action(field, selected_file)
 
     def do_update_flat_or_dark(self, field: Field, selected_file: str) -> None:
-        fg = FilenameGroup.from_file(Path(selected_file))
-        fg.find_all_files()
-        field.set_images(list(fg.all_files()))
+        """
+        Update the field with the images from the selected file.
+        :param  field (Field): The field to update.
+        :param  selected_file (str): The path of the selected file.
+        :return None
+        """
+        filename_group = FilenameGroup.from_file(Path(selected_file))
+        filename_group.find_all_files()
+        field.set_images(list(filename_group.all_files()))
 
     def get_parameters(self) -> LoadingParameters:
         """
-        Gather information from the dialog into a LoadingParameters
+        Get the loading parameters for the selected sample.
+        :return LoadingParameters: The loading parameters for the selected sample.
+        :raises  RuntimeError: If no sample is selected.
         """
         sample_field = self.view.fields[FILE_TYPES.SAMPLE.fname]
         if sample_field.path is None:
             raise RuntimeError("No sample selected")
 
         loading_param = LoadingParameters()
         for file_type in FILE_TYPES:
-            if file_type.mode == "log":
-                continue
-            field = self.view.fields[file_type.fname]
-            if not field.use.isChecked() or field.path is None:
-                continue
-            file_group = FilenameGroup.from_file(field.path)
-            file_group.find_all_files()
-            image_param = ImageParameters(file_group)
-
-            if file_type in log_for_file_type:
-                log_field = self.view.fields[log_for_file_type[file_type].fname]
-                if log_field.use.isChecked():
-                    image_param.log_file = log_field.path
-
-            if file_type == FILE_TYPES.SAMPLE:
-                image_param.indices = field.indices
-
-            loading_param.image_stacks[file_type] = image_param
+            self._update_loading_param(file_type, loading_param)
 
         loading_param.name = sample_field.path.name
         loading_param.pixel_size = self.view.pixelSize.value()
         loading_param.dtype = self.view.pixel_bit_depth.currentText()
         loading_param.sinograms = self.view.images_are_sinograms.isChecked()
         return loading_param
 
+    def _update_loading_param(self, file_type, loading_param):
+        """
+        Update the loading parameters for a specific file type.
+        :param  file_type (str): The type of the file being loaded.
+        :param  loading_param (LoadingParameters): The loading parameters object.
+        :return None
```

Comment:
Can the types be annotations, so `mypy` can catch errors. Same for `_update_image_param`

Reply:
Yes, will modify accordingly